### PR TITLE
Message improvements

### PR DESCRIPTION
**Pull request by rotty**
This addresses #139, at least partially.

### CODE REVIEW

**Comment by rotty** [comment]
@birkenfeld Would you mind having a look?

**Comment by birkenfeld** [comment]
Thanks, I will have time soon.

File: src/message.rs [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+impl fmt::Debug for Message {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self.deref())
+    }
+}
+
+unsafe extern fn drop_msg_content_box(data: *mut c_void, _hint: *mut c_void) {
+    let _ = Box::from_raw(data as *mut u8);
```
Comment (birkenfeld) [inline src/message.rs]: `drop(Box::from_raw(...))` might be a tad clearer.

File: src/message.rs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    /// This is equivalent to using the `From<&[u8]>` trait.
+    pub fn from_slice(data: &[u8]) -> Message {
+        unsafe {
+            let mut msg = Message::with_capacity_unallocated(data.len());
+            ptr::copy_nonoverlapping(data.as_ptr(), msg.as_mut_ptr(), data.len());
+            msg
+        }
+    }
+
+    fn from_box(data: Box<[u8]>) -> Message {
```
Comment (birkenfeld) [inline src/message.rs]: Why not expose this one as well? If you have only a box (which I admit is not that likely) you'd have to copy it into a Vec.

File: src/message.rs [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        Message::from_slice(msg.as_bytes())
+    }
+}
+
+impl<'a> From<&'a String> for Message {
+    /// Construct from a string slice by copying the UTF-8 data.
+    fn from(msg: &String) -> Self {
+        Message::from_slice(msg.as_bytes())
+    }
+}
```
Comment (birkenfeld) [inline src/message.rs]: Add a `From<String>` impl?

**Pull request closed without merging**